### PR TITLE
[ICDS Dashboard] Pull the phone number from the person case ucr

### PR DESCRIPTION
**Pull request by Rohit25negi**
This change is to pull the phone number of the mother directly from the person case UCR instead of from the reference field in the child health UCR. 

I believe, One child_health case cannot be attached to more than 1 person case of the mother so the join should be fine.  This helper already have join with person case ucr. But that is on child person case_id.
Ticket: https://dimagi-dev.atlassian.net/browse/ICDS-1633

I don't see much increase in the Cost:

Current Cost:

```
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.worker_column_276, remote_scan.worker_column_277
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Gather  (cost=34235.27..216303.02 rows=39356 width=3322)
                     Workers Planned: 5
                     ->  Nested Loop Left Join  (cost=33235.27..211367.42 rows=7871 width=3322)
                           ->  Parallel Hash Left Join  (cost=33234.85..206184.85 rows=7871 width=2907)
                                 Hash Cond: ((child_health.state_id = (df.state_id)::text) AND (child_health.doc_id = (df.case_id)::text) AND (child_health.supervisor_id = df.supervisor_id))
                                 ->  Parallel Hash Left Join  (cost=20862.93..193750.96 rows=7871 width=2788)
                                       Hash Cond: ((child_health.state_id = agg_availing.state_id) AND (child_health.mother_id = (agg_availing.person_case_id)::text) AND (child_health.supervisor_id = agg_availing.supervisor_id))
                                       ->  Nested Loop Left Join  (cost=15985.96..188812.00 rows=7871 width=2638)
                                             ->  Nested Loop Left Join  (cost=15985.53..180651.16 rows=7871 width=2521)
                                                   ->  Nested Loop Left Join  (cost=15984.98..167962.17 rows=7871 width=2114)
                                                         ->  Parallel Hash Left Join  (cost=15984.43..149203.34 rows=7871 width=1932)
                                                               Hash Cond: ((child_health.state_id = (thr.state_id)::text) AND (child_health.doc_id = (thr.case_id)::text) AND (child_health.supervisor_id = thr.supervisor_id))
                                                               ->  Nested Loop Left Join  (cost=2686.05..135842.98 rows=7871 width=1815)
                                                                     ->  Nested Loop Left Join  (cost=2685.50..120632.10 rows=7871 width=1684)
                                                                           ->  Nested Loop Left Join  (cost=2685.50..106268.61 rows=7871 width=1135)
                                                                                 ->  Parallel Bitmap Heap Scan on "ucr_icds-cas_static-child_health_cases_a46c129f_103418" child_health  (cost=2685.07..93805.62 rows=7871 width=702)
                                                                                       Recheck Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                       Filter: ((doc_id IS NOT NULL) AND (('2020-07-31'::date - (opened_on)::date) >= 0) AND ((closed = 0) OR (((closed_on)::date - '2020-07-01'::date) > 0)))
                                                                                       ->  Bitmap Index Scan on "ix_ucr_icds-cas_static-child_health_cases_a46c1_120becab_103418"  (cost=0.00..2675.23 rows=134734 width=0)
                                                                                             Index Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                 ->  Index Scan using "ix_ucr_icds-cas_static-child_tasks_cases_3548e5_9585e11e_102906" on "ucr_icds-cas_static-child_tasks_cases_3548e54b_102906" child_tasks  (cost=0.43..1.57 rows=1 width=433)
                                                                                       Index Cond: (child_health.doc_id = child_health_case_id)
                                                                                       Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                           ->  Index Scan using "ix_ucr_icds-cas_person_cases_v3_doc_id_hash_103866" on "ucr_icds-cas_static-person_cases_v3_2ae0879a_103866" person_cases  (cost=0.00..1.81 rows=1 width=549)
                                                                                 Index Cond: (child_health.mother_id = doc_id)
                                                                                 Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                     ->  Index Scan using icds_dashboard_comp_feed_form_pkey_102072 on icds_dashboard_comp_feed_form_102072 cf  (cost=0.55..1.92 rows=1 width=131)
                                                                           Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                                           Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                               ->  Parallel Hash  (cost=13219.79..13219.79 rows=4491 width=117)
                                                                     ->  Parallel Bitmap Heap Scan on icds_dashboard_child_health_thr_forms_102200 thr  (cost=337.94..13219.79 rows=4491 width=117)
                                                                           Recheck Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                                           ->  Bitmap Index Scan on icds_dashboard_child_hea_state_id_month_3050cc30_idx_102200  (cost=0.00..334.46 rows=13921 width=0)
                                                                                 Index Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                         ->  Index Scan using icds_dashboard_growth_monitoring_forms_pkey_102264 on icds_dashboard_growth_monitoring_forms_102264 gm  (cost=0.56..2.37 rows=1 width=182)
                                                               Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                               Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                   ->  Index Scan using icds_dashboard_child_health_postnatal_forms_pkey_102328 on icds_dashboard_child_health_postnatal_forms_102328 pnc  (cost=0.55..1.60 rows=1 width=407)
                                                         Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                         Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                             ->  Index Scan using icds_dashboard_migration_forms_person_case_id_e_2ad5b4a2_368828 on icds_dashboard_migration_forms_368828 agg_migration  (cost=0.42..1.03 rows=1 width=117)
                                                   Index Cond: (child_health.mother_id = (person_case_id)::text)
                                                   Filter: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text) AND (child_health.supervisor_id = supervisor_id))
                                       ->  Parallel Hash  (cost=4810.25..4810.25 rows=3813 width=150)
                                             ->  Parallel Seq Scan on icds_dashboard_availing_service_forms_495484 agg_availing  (cost=0.00..4810.25 rows=3813 width=150)
                                                   Filter: ((month = '2020-07-01'::date) AND (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                                 ->  Parallel Hash  (cost=12208.24..12208.24 rows=9353 width=119)
                                       ->  Parallel Index Scan using icds_dashboard_daily_feeding_forms_month_state__cdb5a544_102584 on icds_dashboard_daily_feeding_forms_102584 df  (cost=0.56..12208.24 rows=9353 width=119)
                                             Index Cond: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                           ->  Index Scan using "ix_ucr_icds-test_static-child_delivery_forms_sup_case_id_103290" on "ucr_icds-cas_static-child_delivery_forms_eafd4095_103290" del_form  (cost=0.42..0.65 rows=1 width=349)
                                 Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = child_health_case_id))
(58 rows)

```

New cost:

```
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.worker_column_336, remote_scan.worker_column_337
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Gather  (cost=34235.27..230666.51 rows=39356 width=3871)
                     Workers Planned: 5
                     ->  Nested Loop Left Join  (cost=33235.27..225730.91 rows=7871 width=3871)
                           ->  Parallel Hash Left Join  (cost=33234.85..220548.34 rows=7871 width=3456)
                                 Hash Cond: ((child_health.state_id = (df.state_id)::text) AND (child_health.doc_id = (df.case_id)::text) AND (child_health.supervisor_id = df.supervisor_id))
                                 ->  Parallel Hash Left Join  (cost=20862.93..208114.44 rows=7871 width=3337)
                                       Hash Cond: ((child_health.state_id = agg_availing.state_id) AND (child_health.mother_id = (agg_availing.person_case_id)::text) AND (child_health.supervisor_id = agg_availing.supervisor_id))
                                       ->  Nested Loop Left Join  (cost=15985.96..203175.48 rows=7871 width=3187)
                                             ->  Nested Loop Left Join  (cost=15985.53..195014.64 rows=7871 width=3070)
                                                   ->  Nested Loop Left Join  (cost=15984.98..182325.66 rows=7871 width=2663)
                                                         ->  Parallel Hash Left Join  (cost=15984.43..163566.83 rows=7871 width=2481)
                                                               Hash Cond: ((child_health.state_id = (thr.state_id)::text) AND (child_health.doc_id = (thr.case_id)::text) AND (child_health.supervisor_id = thr.supervisor_id))
                                                               ->  Nested Loop Left Join  (cost=2686.05..150206.46 rows=7871 width=2364)
                                                                     ->  Nested Loop Left Join  (cost=2685.50..134995.58 rows=7871 width=2233)
                                                                           ->  Nested Loop Left Join  (cost=2685.50..120632.10 rows=7871 width=1684)
                                                                                 ->  Nested Loop Left Join  (cost=2685.50..106268.61 rows=7871 width=1135)
                                                                                       ->  Parallel Bitmap Heap Scan on "ucr_icds-cas_static-child_health_cases_a46c129f_103418" child_health  (cost=2685.07..93805.62 rows=7871 width=702)
                                                                                             Recheck Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                             Filter: ((doc_id IS NOT NULL) AND (('2020-07-31'::date - (opened_on)::date) >= 0) AND ((closed = 0) OR (((closed_on)::date - '2020-07-01'::date) > 0)))
                                                                                             ->  Bitmap Index Scan on "ix_ucr_icds-cas_static-child_health_cases_a46c1_120becab_103418"  (cost=0.00..2675.23 rows=134734 width=0)
                                                                                                   Index Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                       ->  Index Scan using "ix_ucr_icds-cas_static-child_tasks_cases_3548e5_9585e11e_102906" on "ucr_icds-cas_static-child_tasks_cases_3548e54b_102906" child_tasks  (cost=0.43..1.57 rows=1 width=433)
                                                                                             Index Cond: (child_health.doc_id = child_health_case_id)
                                                                                             Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                                 ->  Index Scan using "ix_ucr_icds-cas_person_cases_v3_doc_id_hash_103866" on "ucr_icds-cas_static-person_cases_v3_2ae0879a_103866" person_cases  (cost=0.00..1.81 rows=1 width=549)
                                                                                             Filter: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                           ->  Index Scan using "ix_ucr_icds-cas_static-child_tasks_cases_3548e5_9585e11e_102906" on "ucr_icds-cas_static-child_tasks_cases_3548e54b_102906" child_tasks  (cost=0.43..1.57 rows=1 width=433)
                                                                                 Index Cond: (child_health.doc_id = child_health_case_id)
                                                                                 Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                     ->  Index Scan using icds_dashboard_comp_feed_form_pkey_102072 on icds_dashboard_comp_feed_form_102072 cf  (cost=0.55..1.92 rows=1 width=131)
                                                                           Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                                           Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                               ->  Parallel Hash  (cost=13219.79..13219.79 rows=4491 width=117)
                                                                     ->  Parallel Bitmap Heap Scan on icds_dashboard_child_health_thr_forms_102200 thr  (cost=337.94..13219.79 rows=4491 width=117)
                                                                           Recheck Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                                           ->  Bitmap Index Scan on icds_dashboard_child_hea_state_id_month_3050cc30_idx_102200  (cost=0.00..334.46 rows=13921 width=0)
                                                                                 Index Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                         ->  Index Scan using icds_dashboard_growth_monitoring_forms_pkey_102264 on icds_dashboard_growth_monitoring_forms_102264 gm  (cost=0.56..2.37 rows=1 width=182)
                                                               Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                               Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                   ->  Index Scan using icds_dashboard_child_health_postnatal_forms_pkey_102328 on icds_dashboard_child_health_postnatal_forms_102328 pnc  (cost=0.55..1.60 rows=1 width=407)
                                                         Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                         Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                             ->  Index Scan using icds_dashboard_migration_forms_person_case_id_e_2ad5b4a2_368828 on icds_dashboard_migration_forms_368828 agg_migration  (cost=0.42..1.03 rows=1 width=117)
                                                   Index Cond: (child_health.mother_id = (person_case_id)::text)
                                                   Filter: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text) AND (child_health.supervisor_id = supervisor_id))
                                       ->  Parallel Hash  (cost=4810.25..4810.25 rows=3813 width=150)
                                             ->  Parallel Seq Scan on icds_dashboard_availing_service_forms_495484 agg_availing  (cost=0.00..4810.25 rows=3813 width=150)
                                                   Filter: ((month = '2020-07-01'::date) AND (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                                 ->  Parallel Hash  (cost=12208.24..12208.24 rows=9353 width=119)
                                       ->  Parallel Index Scan using icds_dashboard_daily_feeding_forms_month_state__cdb5a544_102584 on icds_dashboard_daily_feeding_forms_102584 df  (cost=0.56..12208.24 rows=9353 width=119)
                                             Index Cond: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                           ->  Index Scan using "ix_ucr_icds-test_static-child_delivery_forms_sup_case_id_103290" on "ucr_icds-cas_static-child_delivery_forms_eafd4095_103290" del_form  (cost=0.42..0.65 rows=1 width=349)
                                 Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = child_health_case_id))
(64 rows)

```

### CODE REVIEW

**Comment by Rohit25negi** [comment]
Acutal query cost:



current query:


```
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.supervisor_id, remote_scan.awc_id
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Gather  (cost=36161.85..243002.75 rows=39345 width=890)
                     Workers Planned: 5
                     ->  Nested Loop Left Join  (cost=35161.85..238068.25 rows=7869 width=890)
                           ->  Parallel Hash Left Join  (cost=35161.43..208117.75 rows=7869 width=867)
                                 Hash Cond: ((child_health.state_id = (df.state_id)::text) AND (child_health.doc_id = (df.case_id)::text) AND (child_health.su
pervisor_id = df.supervisor_id))
                                 ->  Parallel Hash Left Join  (cost=22573.05..195467.40 rows=7869 width=863)
                                       Hash Cond: ((child_health.state_id = agg_availing.state_id) AND (child_health.mother_id = (agg_availing.person_case_id)
::text) AND (child_health.supervisor_id = agg_availing.supervisor_id))
                                       ->  Nested Loop Left Join  (cost=17691.26..190523.64 rows=7869 width=853)
                                             ->  Nested Loop Left Join  (cost=17690.84..182363.79 rows=7869 width=843)
                                                   ->  Nested Loop Left Join  (cost=17690.29..169671.37 rows=7869 width=543)
                                                         ->  Parallel Hash Left Join  (cost=17689.73..150915.35 rows=7869 width=476)
                                                               Hash Cond: ((child_health.state_id = (thr.state_id)::text) AND (child_health.doc_id = (thr.case
_id)::text) AND (child_health.supervisor_id = thr.supervisor_id))
                                                               ->  Nested Loop Left Join  (cost=2687.15..135850.80 rows=7869 width=474)
                                                                     ->  Nested Loop Left Join  (cost=2686.60..120640.83 rows=7869 width=458)
                                                                           ->  Nested Loop Left Join  (cost=2686.60..106280.36 rows=7869 width=441)
                                                                                 ->  Parallel Bitmap Heap Scan on "ucr_icds-cas_static-child_health_cases_a46c
129f_103418" child_health  (cost=2686.17..93810.58 rows=7869 width=317)
                                                                                       Recheck Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                       Filter: ((doc_id IS NOT NULL) AND (('2020-07-31'::date - (opened_on)::d
ate) >= 0) AND ((closed = 0) OR (((closed_on)::date - '2020-07-01'::date) > 0)))
                                                                                       ->  Bitmap Index Scan on "ix_ucr_icds-cas_static-child_health_cases_a46
c1_120becab_103418"  (cost=0.00..2676.33 rows=134734 width=0)
                                                                                             Index Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                 ->  Index Scan using "ix_ucr_icds-cas_static-child_tasks_cases_3548e5_9585e11
e_102906" on "ucr_icds-cas_static-child_tasks_cases_3548e54b_102906" child_tasks  (cost=0.43..1.57 rows=1 width=227)
                                                                                       Index Cond: (child_health.doc_id = child_health_case_id)
                                                                                       Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (chi
ld_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                           ->  Index Scan using "ix_ucr_icds-cas_person_cases_v3_doc_id_hash_103866" on "ucr_i
cds-cas_static-person_cases_v3_2ae0879a_103866" person_cases  (cost=0.00..1.81 rows=1 width=120)
                                                                                 Index Cond: (child_health.mother_id = doc_id)
                                                                                 Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                     ->  Index Scan using icds_dashboard_comp_feed_form_pkey_102072 on icds_dashboard_comp_feed_form_102072 cf  (cost=0.55..1.92 rows=1 width=119)
                                                                           Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                                           Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                               ->  Parallel Hash  (cost=14912.05..14912.05 rows=5173 width=105)
                                                                     ->  Parallel Bitmap Heap Scan on icds_dashboard_child_health_thr_forms_102200 thr  (cost=388.23..14912.05 rows=5173 width=105)
                                                                           Recheck Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                                           ->  Bitmap Index Scan on icds_dashboard_child_hea_state_id_month_3050cc30_idx_102200  (cost=0.00..384.22 rows=16037 width=0)
                                                                                 Index Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                         ->  Index Scan using icds_dashboard_growth_monitoring_forms_pkey_102264 on icds_dashboard_growth_monitoring_forms_102264 gm  (cost=0.56..2.37 rows=1 width=170)
                                                               Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                               Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                   ->  Index Scan using icds_dashboard_child_health_postnatal_forms_pkey_102328 on icds_dashboard_child_health_postnatal_forms_102328 pnc  (cost=0.55..1.60 rows=1 width=403)
                                                         Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                         Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                             ->  Index Scan using icds_dashboard_migration_forms_person_case_id_e_2ad5b4a2_368828 on icds_dashboard_migration_forms_368828 agg_migration  (cost=0.42..1.03 rows=1 width=113)
                                                   Index Cond: (child_health.mother_id = (person_case_id)::text)
                                                   Filter: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text) AND (child_health.supervisor_id = supervisor_id))
                                       ->  Parallel Hash  (cost=4814.91..4814.91 rows=3822 width=113)
                                             ->  Parallel Seq Scan on icds_dashboard_availing_service_forms_495484 agg_availing  (cost=0.00..4814.91 rows=3822 width=113)
                                                   Filter: ((month = '2020-07-01'::date) AND (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                                 ->  Parallel Hash  (cost=12423.20..12423.20 rows=9439 width=107)
                                       ->  Parallel Index Scan using icds_dashboard_daily_feeding_forms_month_state__cdb5a544_102584 on icds_dashboard_daily_feeding_forms_102584 df  (cost=0.56..12423.20 rows=9439 width=107)
                                             Index Cond: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                           ->  Index Scan using "ix_ucr_icds-test_static-child_delivery_forms_sup_case_id_103290" on "ucr_icds-cas_static-child_delivery_forms_eafd4095_103290" del_form  (cost=0.42..0.65 rows=1 width=86)
                                 Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = child_health_case_id))
(58 rows)

```



Updated query with this change:

```
 Sort  (cost=0.00..0.00 rows=0 width=0)
   Sort Key: remote_scan.supervisor_id, remote_scan.awc_id
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Gather  (cost=36161.85..257363.22 rows=39345 width=891)
                     Workers Planned: 5
                     ->  Nested Loop Left Join  (cost=35161.85..252428.72 rows=7869 width=891)
                           ->  Parallel Hash Left Join  (cost=35161.43..222478.23 rows=7869 width=868)
                                 Hash Cond: ((child_health.state_id = (df.state_id)::text) AND (child_health.doc_id = (df.case_id)::text) AND (child_health.supervisor_id = df.supervisor_id))
                                 ->  Parallel Hash Left Join  (cost=22573.05..209827.88 rows=7869 width=864)
                                       Hash Cond: ((child_health.state_id = agg_availing.state_id) AND (child_health.mother_id = (agg_availing.person_case_id)::text) AND (child_health.supervisor_id = agg_availing.supervisor_id))
                                       ->  Nested Loop Left Join  (cost=17691.26..204884.12 rows=7869 width=854)
                                             ->  Nested Loop Left Join  (cost=17690.84..196724.27 rows=7869 width=844)
                                                   ->  Nested Loop Left Join  (cost=17690.29..184031.85 rows=7869 width=544)
                                                         ->  Parallel Hash Left Join  (cost=17689.73..165275.83 rows=7869 width=477)
                                                               Hash Cond: ((child_health.state_id = (thr.state_id)::text) AND (child_health.doc_id = (thr.case_id)::text) AND (child_health.supervisor_id = thr.supervisor_id))
                                                               ->  Nested Loop Left Join  (cost=2687.15..150211.28 rows=7869 width=475)
                                                                     ->  Nested Loop Left Join  (cost=2686.60..135001.31 rows=7869 width=459)
                                                                           ->  Nested Loop Left Join  (cost=2686.60..120640.83 rows=7869 width=458)
                                                                                 ->  Nested Loop Left Join  (cost=2686.60..106280.36 rows=7869 width=441)
                                                                                       ->  Parallel Bitmap Heap Scan on "ucr_icds-cas_static-child_health_cases_a46c129f_103418" child_health  (cost=2686.17..93810.58 rows=7869 width=317)
                                                                                             Recheck Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                             Filter: ((doc_id IS NOT NULL) AND (('2020-07-31'::date - (opened_on)::date) >= 0) AND ((closed = 0) OR (((closed_on)::date - '2020-07-01'::date) > 0)))
                                                                                             ->  Bitmap Index Scan on "ix_ucr_icds-cas_static-child_health_cases_a46c1_120becab_103418"  (cost=0.00..2676.33 rows=134734 width=0)
                                                                                                   Index Cond: (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text)
                                                                                       ->  Index Scan using "ix_ucr_icds-cas_static-child_tasks_cases_3548e5_9585e11e_102906" on "ucr_icds-cas_static-child_tasks_cases_3548e54b_102906" child_tasks  (cost=0.43..1.57 rows=1 width=227)
                                                                                             Index Cond: (child_health.doc_id = child_health_case_id)
                                                                                             Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                                 ->  Index Scan using "ix_ucr_icds-cas_person_cases_v3_doc_id_hash_103866" on "ucr_icds-cas_static-person_cases_v3_2ae0879a_103866" person_cases  (cost=0.00..1.81 rows=1 width=120)
                                                                                       Index Cond: (child_health.mother_id = doc_id)
                                                                                       Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                           ->  Index Scan using "ix_ucr_icds-cas_person_cases_v3_doc_id_hash_103866" on "ucr_icds-cas_static-person_cases_v3_2ae0879a_103866" mother_person_cases  (cost=0.00..1.81 rows=1 width=104)
                                                                                 Index Cond: (child_health.mother_case_id = doc_id)
                                                                                 Filter: ((state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = state_id) AND (child_health.supervisor_id = supervisor_id))
                                                                     ->  Index Scan using icds_dashboard_comp_feed_form_pkey_102072 on icds_dashboard_comp_feed_form_102072 cf  (cost=0.55..1.92 rows=1 width=119)
                                                                           Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                                           Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                               ->  Parallel Hash  (cost=14912.05..14912.05 rows=5173 width=105)
                                                                     ->  Parallel Bitmap Heap Scan on icds_dashboard_child_health_thr_forms_102200 thr  (cost=388.23..14912.05 rows=5173 width=105)
                                                                           Recheck Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                                           ->  Bitmap Index Scan on icds_dashboard_child_hea_state_id_month_3050cc30_idx_102200  (cost=0.00..384.22 rows=16037 width=0)
                                                                                 Index Cond: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (month = '2020-07-01'::date))
                                                         ->  Index Scan using icds_dashboard_growth_monitoring_forms_pkey_102264 on icds_dashboard_growth_monitoring_forms_102264 gm  (cost=0.56..2.37 rows=1 width=170)
                                                               Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                               Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                                   ->  Index Scan using icds_dashboard_child_health_postnatal_forms_pkey_102328 on icds_dashboard_child_health_postnatal_forms_102328 pnc  (cost=0.55..1.60 rows=1 width=403)
                                                         Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = (case_id)::text) AND (month = '2020-07-01'::date))
                                                         Filter: (((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text))
                                             ->  Index Scan using icds_dashboard_migration_forms_person_case_id_e_2ad5b4a2_368828 on icds_dashboard_migration_forms_368828 agg_migration  (cost=0.42..1.03 rows=1 width=113)
                                                   Index Cond: (child_health.mother_id = (person_case_id)::text)
                                                   Filter: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text) AND (child_health.state_id = (state_id)::text) AND (child_health.supervisor_id = supervisor_id))
                                       ->  Parallel Hash  (cost=4814.91..4814.91 rows=3822 width=113)
                                             ->  Parallel Seq Scan on icds_dashboard_availing_service_forms_495484 agg_availing  (cost=0.00..4814.91 rows=3822 width=113)
                                                   Filter: ((month = '2020-07-01'::date) AND (state_id = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                                 ->  Parallel Hash  (cost=12423.20..12423.20 rows=9439 width=107)
                                       ->  Parallel Index Scan using icds_dashboard_daily_feeding_forms_month_state__cdb5a544_102584 on icds_dashboard_daily_feeding_forms_102584 df  (cost=0.56..12423.20 rows=9439 width=107)
                                             Index Cond: ((month = '2020-07-01'::date) AND ((state_id)::text = 'f98e91aa003accb7b849a0f18ebd7039'::text))
                           ->  Index Scan using "ix_ucr_icds-test_static-child_delivery_forms_sup_case_id_103290" on "ucr_icds-cas_static-child_delivery_forms_eafd4095_103290" del_form  (cost=0.42..0.65 rows=1 width=86)
                                                                                                           Index Cond: ((child_health.supervisor_id = supervisor_id) AND (child_health.doc_id = child_health_case_id))
(62 rows)
```

File: custom/icds_reports/reports/awc_reports.py [hunk 1]
```diff
@@ -1063,7 +1064,7 @@ def base_data(row_data):
                 data_entered=True if row_data.recorded_height and row_data.recorded_weight else False
             ),
             pse_days_attended=row_data.pse_days_attended,
-            mother_phone_number=row_data.mother_phone_number,
+            mother_phone_number=phone_number_function(row_data.mother_phone_number),
             aww_phone_number=row_data.aww_phone_number,
```
Comment (calellowitz) [inline custom/icds_reports/reports/awc_reports.py:1068]: does this need the `phone_number_function` as well?

**Review by calellowitz — APPROVED** [review]
Looks good. would be great to make sure this code is exercised by tests

**Comment by Rohit25negi** [comment]
merging it

**Pull request merged**